lcls-twincat-sample-delivery/Library/POUs/VICI/FB_4PViciValve.TcPOU

```diff
@@ -0,0 +1,68 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4024.12">
+  <POU Name="FB_4PViciValve" Id="{867cd0ab-cbb3-428d-92e9-f85cb5a24957}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_4PViciValve
+VAR_INPUT
+    iAddress : INT;
+END_VAR
+VAR_OUTPUT
+END_VAR
+VAR
+
+    // Re-Use Vici Driver Code for 12 port Value
+    fbViciDriver  : FB_ViciDriver;
+    fbViciSerial  : FB_SerialCommWrapper;
+    stViciSerial  : ST_SerialComm;
+    stViciVlvCtrl : ST_ViciControl;
+    stViciVlvStatus : ST_ViciStatus;
+    //
+    iNewPos: INT;
+    iMaped : INT;
+
+
+    {attribute 'pytmc' := '
+        pv: ReqPos
+        io: io
+    '}
+    iVici1ReqPos : INT;
+
+    {attribute 'pytmc' := '
+        pv: SampleNames
+        io: io
+     '}
+     stSampleName : ST_4PViciValveSampleNames;
+
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[// Re-using 12 Port Code, Need to map 1-4 to vlaues that 12 port Commands
+//map 1-4 to 9 - 12;
```

Comment:
What's the story behind this mapping? Where's the documentation you read to decide on this mapping

Reply:
There Is a code block in the [FB_VICIDriver ](https://github.com/pcdshub/lcls-twincat-sample-delivery/blob/2eab9f0c6295b374a5f2c6be21e9916c6d16f390/lcls-twincat-sample-delivery/Library/POUs/VICI/FB_ViciDriver.TcPOU#L94).  

There is a conversion that comments (* Mirror the "clock" value across the vertical axis of symmetry *). This means that the port requested does not equal to the command sent to the valve. Requesting port 1 sends command  for the port to go to the 12th port.  The mapping used in the 4port FB to match the requested port number with the appropriate command for the 4 port valve.